src/Build/Logging/ParallelLogger/ParallelConsoleLogger.cs

```diff
@@ -205,6 +205,8 @@ internal override void ResetConsoleLoggerState()
             _hasBuildStarted = false;
 
             // Reset the two data structures created when the logger was created
+            target_framework_mapping = new Dictionary<(int, int), string>();
```

Comment:
If this is parallel, don't these have to be thread-safe?

Reply:
Yep. I changed them to ConcurrentDictionary which I believe is thread safe.